File: java/SecureSessionTimeout.java
//Concept: Explicit session TTL
public class SecureSessionTimeout {

    static long createdAt = System.currentTimeMillis();
    static long TTL = 5000;

    static boolean isValid() {
        return System.currentTimeMillis() - createdAt < TTL;
    }

    public static void main(String[] args) throws Exception {
        Thread.sleep(3000);
        System.out.println("Session valid: " + isValid());
    }
}

File: python/secure_session_timeout.py
#Concept: Proper session expiry check
import time

class Session:
    def __init__(self, ttl):
        self.created_at = time.time()
        self.ttl = ttl

    def is_valid(self):
        return (time.time() - self.created_at) < self.ttl

s = Session(5)
time.sleep(3)
print("Session valid:", s.is_valid())
